pkg/router/template/plugin.go

```diff
@@ -254,42 +254,62 @@ func getPartsFromEndpointsKey(key ServiceUnitKey) (string, string) {
 	return namespace, name
 }
 
+// subsetHasAddresses returns true if subsets has any addresses.
+func subsetHasAddresses(subsets []kapi.EndpointSubset) bool {
+	for i := range subsets {
+		if len(subsets[i].Addresses) > 0 {
+			return true
+		}
+	}
+	return false
+}
+
+// serviceIsIdled return true if the service has been annotated with
+// unidlingapi.IdledAtAnnotation.
+func serviceIsIdled(service *kapi.Service) bool {
+	value, _ := service.Annotations[unidlingapi.IdledAtAnnotation]
+	return len(value) > 0
+}
+
 // createRouterEndpoints creates openshift router endpoints based on k8s endpoints
 func createRouterEndpoints(endpoints *kapi.Endpoints, excludeUDP bool, lookupSvc ServiceLookup) []Endpoint {
 	// check if this service is currently idled
 	wasIdled := false
 	subsets := endpoints.Subsets
-	if _, ok := endpoints.Annotations[unidlingapi.IdledAtAnnotation]; ok && len(endpoints.Subsets) == 0 {
+
+	if !subsetHasAddresses(subsets) {
```

Comment:
If `!subsetHasAddresses(subsets)` evaluates to `true`, does that guarantee that the service is being idled? I ask because the current error wrapper for `lookupSvc.LookupService(endpoints)` seems to assume so. Maybe that error message should be updated to make more sense?

Reply:
Your suggestion is to drop the word "idled" here?

```go
"unable to find idled service corresponding to idled endpoints %s/%s: %v"
```

I ask because I did that first time, then decided to not to... but it did seem like the natural change at the time.

Reply:
Yea, I think it would make sense to drop the word "idled" in both instances here. Any objections?

Reply:
Took a punt in https://github.com/openshift/router/pull/225/commits/b873ba19babcad05914a27ca2b4768b0c26b062e.

I think the notion of idle is irrelevant - I looked up a service and I don't have it.